### PR TITLE
Disable profiler-related features for OpenJ9 JVM

### PR DESCRIPTION
**Pull request by jeanbisutti**
Disable profiler-related features for OpenJ9 JVM

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 1]
```diff
@@ -243,13 +243,23 @@ private static void logWarningIfUsingInternalAttributes(String attributeKey) {
   }
 
   private static void overlayProfilerEnvVars(Configuration config) {
+    if (isOpenJ9Jvm()) {
+      configurationLogger.warn(
+          "Profiler is not supported for an OpenJ9 JVM. Instead, please use an OpenJDK JVM.");
+      return;
+    }
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:250]: this won't update `profiler.enabled` if it was set via boolean, how about moving this check (and updating the boolean) near the end of `create` method?

Comment (jeanbisutti) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:250]: The `profiler.enabled` will keep its default value, that is to say false. I would prefer to have all the code related to  `profiler.enabled`  intialization in only one place.

**Review by trask — APPROVED** [review]

**Comment by jeanbisutti** [comment]
I have added a commit to make clear that the creation of the configuration has 3 steps:

```
public static Configuration create(Path agentJarPath, @Nullable RpConfiguration rpConfiguration)
      throws IOException {
    Configuration config = loadConfigurationFile(agentJarPath);
    logConfigurationWarnings(config);
    overlayConfiguration(agentJarPath, rpConfiguration, config);
    return config;
  }
```

**Comment by trask** [comment]
> I have added a commit to make clear that the creation of the configuration has 3 steps:

nice!

can you review #2463, so that I can merge that as it may have conflicts with this PR?

**Comment by jeanbisutti** [comment]
> > I have added a commit to make clear that the creation of the configuration has 3 steps:
> 
> nice!
> 
> can you review #2463, so that I can merge that as it may have conflicts with this PR?

@trask Done

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java [hunk 2]
```diff
@@ -488,7 +506,7 @@ static void overlayFromEnv(Configuration config, Path baseDir) throws IOExceptio
     loadJmxMetricsEnvVar(config);
 
     addDefaultJmxMetricsIfNotPresent(config);
-    overlayProfilerEnvVars(config);
+    overlayProfilerConfig(config);
     overlayAadEnvVars(config);
     overlayInstrumentationEnabledEnvVars(config);
```
Comment (trask) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/ConfigurationBuilder.java:511]: why change ...EnvVars to ...Config?

maybe `overlayProfilerConfigFromEnv`?

**Review by trask — APPROVED** [review]

**Pull request merged**